src/main/java/pw/twpi/whitelistsync2/services/PostgreSqlService.java

```diff
@@ -0,0 +1,290 @@
+				}
+				sql.append(String.format("(%s, %s, %s)", wlp.getUuid(), wlp.getName(), wlp.isWhitelisted() ? "true" : "false"));
+				isNotFirst = true;
+			}
+			stmt.executeUpdate(sql.toString());
+			return true;
+		} catch (SQLException e) {
+			WhitelistSync2.LOGGER.error("Unexpected exception while copying local whitelist to database", e);
+		}
+		return false;
+	}
+
+	@Override
+	public boolean copyLocalOppedPlayersToDatabase() {
+		if (!Config.SYNC_OP_LIST.get()) {
+			WhitelistSync2.LOGGER.error("Op list syncing is currently disabled in your config. Please enable it and restart the server to use this feature.");
+			return false;
+		}
+		try (Connection connection = getConnection();
+		     Statement stmt = connection.createStatement()) {
+			ArrayList<OppedPlayer> whitelistedPlayers = getOppedPlayersFromLocal();
+			StringBuilder sql = new StringBuilder("INSERT INTO public.oplist(uuid, playername, is_opped) VALUES ");
+			boolean isNotFirst = false;
+			for (OppedPlayer wlp : whitelistedPlayers) {
+				if (isNotFirst) {
+					sql.append(", ");
+				}
+				sql.append(String.format("(%s, %s, %s)", wlp.getUuid(), wlp.getName(), wlp.isOp() ? "true" : "false"));
+				isNotFirst = true;
+			}
+			stmt.executeUpdate(sql.toString());
+			stmt.close();
+			return true;
+		} catch (SQLException e) {
+			WhitelistSync2.LOGGER.error("Unexpected exception while copying local oplist to database", e);
+		}
+		return false;
+	}
+
+	@Override
+	public boolean copyDatabaseWhitelistedPlayersToLocal(MinecraftServer server) {
+		try (Connection connection = getConnection();
+			 Statement statement = connection.createStatement();
+			 ResultSet result = statement.executeQuery("SELECT uuid, playername, is_whitelisted FROM public.whitelist")) {
+			ArrayList<WhitelistedPlayer> localWhitelistedPlayers = WhitelistedPlayersFileUtilities.getWhitelistedPlayers();
+			while (result.next()) {
+				String uuid = result.getString("uuid");
+				GameProfile player = new GameProfile(UUID.fromString(uuid), result.getString("playername"));
+				if (result.getBoolean("is_whitelisted") && localWhitelistedPlayers.stream().noneMatch(o -> o.getUuid().equals(uuid))) {
+					server.getPlayerList().getWhiteList().add(new WhitelistEntry(player));
+				} else if (localWhitelistedPlayers.stream().anyMatch(o -> o.getUuid().equals(uuid))) {
+					server.getPlayerList().getWhiteList().remove(new WhitelistEntry(player));
+				}
+			}
+		} catch (SQLException e) {
+			WhitelistSync2.LOGGER.error("Unexpected exception while copying whitelist to local", e);
+		}
+		return false;
+	}
+
+	@Override
+	public boolean copyDatabaseOppedPlayersToLocal(MinecraftServer server) {
+		if (!Config.SYNC_OP_LIST.get()) {
+			return false;Text is available under the Creative Commons Attribution-ShareAlike License; additional terms may apply. By using this site, you agree to the Terms of Use and Privacy Policy. Wikipedia® is a registered trademark of the Wikimedia Foundation, Inc., a non-profit organization.
```

Comment:
Is this text here by accident?

Reply:
Yes. I'm not sure how I managed to commit this, as it doesn't compile with it.